middleware/dnstap/setup.go

```diff
@@ -0,0 +1,61 @@
+package dnstap
+
+import (
+	"strconv"
+
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/middleware"
+	"github.com/coredns/coredns/middleware/dnstap/out"
+
+	"github.com/mholt/caddy"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	caddy.RegisterPlugin("dnstap", caddy.Plugin{
+		ServerType: "dns",
+		Action:     wrapSetup,
+	})
+}
+
+func wrapSetup(c *caddy.Controller) error {
+	if err := setup(c); err != nil {
+		return middleware.Error("dnstap", err)
+	}
+	return nil
+}
+
+func setup(c *caddy.Controller) error {
```

Comment:
We should probably talk about how the *dnstap* directive should look in the Corefile. Can you send a proposal?